crud.py

```diff
@@ -0,0 +1,164 @@
+    )
+    db.add(new_author)
+    db.commit()
+    db.refresh(new_author)
+    return new_author
+
+
+def get_author_by_name(db: Session, authorname: str):
+    return db.query(models.DBAuthor).filter(
+        models.DBAuthor.name == authorname
+        ).first()
+
+
+def get_book(db: Session, book_id: int):
+    return db.query(models.DBBook).filter(
+        models.DBBook.id == book_id
+        ).first()
+
+
+def get_book_list(
+    db: Session,
+    skip: int = 0,
+    limit: int = 10,
+    author_id: int | None = None,
+    book_title: str | None = None,
+    sort_field: str | None = None
+):
+    queryset = db.query(models.DBBook)
+    if author_id is not None:
+        queryset = queryset.filter(models.DBBook.author_id == author_id)
+    if book_title is not None:
+        queryset = queryset.filter(models.DBBook.title.contains(book_title))
+    if sort_field is not None:
+        if sort_field.startswith("-"):
+            sort_field = sort_field[1:] + " desc"
+        queryset = queryset.order_by(text(sort_field))
+    return queryset.offset(skip).limit(limit).all()
+
+
+def create_book(db: Session, book: schemas.BookCreate):
+    new_book = models.DBBook(**book.dict())
+    db.add(new_book)
+    db.commit()
+    db.refresh(new_book)
+    return new_book
+
+
+def get_book_by_title(db: Session, title: str):
+    return db.query(models.DBBook).filter(models.DBBook.title == title).first()
+
+
+def patch_book(
+    book_id: int,
+    db: Session,
+    book: schemas.BookCreate
+):
+    db_book = get_book(db, book_id=book_id)
+    if db_book is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Book was not found"
+        )
+    book_data = book.dict(exclude_unset=True)
+    for key, val in book_data.items():
+        setattr(db_book, key, val)
+    db.add(db_book)
+    db.commit()
+    db.refresh(db_book)
+    return db_book
+
+
+def delete_book(book_id: int, db: Session):
+    db_book = get_book(db, book_id=book_id)
+    if db_book is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Book was not found"
+        )
+    db.delete(db_book)
+    db.commit()
+    return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+def patch_author(
+    author_id: int,
+    db: Session,
+    author: schemas.AuthorCreate
+):
+    db_author = get_author(db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Author was not found"
+        )
+    author_data = author.dict(exclude_unset=True)
+    for key, val in author_data.items():
+        setattr(db_author, key, val)
+    db.add(db_author)
+    db.commit()
+    db.refresh(db_author)
+    return db_author
+
+
+def delete_author(author_id: int, db: Session):
+    db_author = get_author(db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Author was not found"
+        )
```

Comment:
seems like I've seen this piece of code before in you app:))
DRY

Reply:
i see, so i just can move code with raising 404 into get_author func? ok ill try